src/charm.py

```diff
@@ -257,59 +258,69 @@ def get_node_ports(self) -> Dict[str, Any]:
     # =======================
 
     def patch_port(self, use_node_port: bool = False) -> None:
-        """Patch Juju-created k8s service.
-
-        The k8s service will be tied to pod-0 so that the service is auto cleaned by
-        k8s when the last pod is scaled down.
-        """
+        """Create or delete a nodeport service for external node connectivity."""
         if not self.unit.is_leader():
             return
 
-        try:
-            logger.debug("Patching k8s service")
-            client = lightkube.Client()
+        client = lightkube.Client()
+        service_name = f"{self.app.name}-nodeport"
+        if use_node_port:
+            logger.debug("Creating nodeport service")
             pod0 = client.get(
                 res=lightkube.resources.core_v1.Pod,
                 name=self.app.name + "-0",
-                namespace=self.model.name,
+                namespace=self._namespace,
             )
             service = lightkube.resources.core_v1.Service(
+                apiVersion="v1",
+                kind="Service",
                 metadata=lightkube.models.meta_v1.ObjectMeta(
-                    name=self.app.name,
-                    namespace=self.model.name,
+                    name=service_name,
+                    namespace=self._namespace,
                     ownerReferences=pod0.metadata.ownerReferences,
                     labels={
                         "app.kubernetes.io/name": self.app.name,
                     },
                 ),
                 spec=lightkube.models.core_v1.ServiceSpec(
+                    selector={"app.kubernetes.io/name": self.app.name},
                     ports=[
                         lightkube.models.core_v1.ServicePort(
                             name="pgbouncer",
                             port=self.config["listen_port"],
                             targetPort=self.config["listen_port"],
                         )
                     ],
-                    type=("NodePort" if use_node_port else "ClusterIP"),
-                    selector={"app.kubernetes.io/name": self.app.name},
+                    type="NodePort",
                 ),
             )
-            client.patch(
-                res=lightkube.resources.core_v1.Service,
-                obj=service,
-                name=service.metadata.name,
-                namespace=service.metadata.namespace,
-                force=True,
-                field_manager=self.app.name,
-                patch_type=lightkube.types.PatchType.MERGE,
-            )
-            logger.debug("Patched k8s service")
```

Comment:
Should we keep on patching the ports on the juju service?

Reply:
I don't think so. I remember Mykola told me that etcd is sensible to write.